src/stability_sdk/client.py

```diff
@@ -104,30 +119,147 @@ def process_artifacts_from_answers(
         self.stub = stub
         self.engine_id = engine_id
 
+class Project():
+    def __init__(self, api: 'Api', project: project.Project):
+        self.api = api
+        self.id = project.id
+        self.file_id = project.file.id
+        self.title = project.title
+
+    @staticmethod
+    def create(
+        api: 'Api', 
+        title: str, 
+        access: project.ProjectAccess=project.PROJECT_ACCESS_PRIVATE,
+        status: project.ProjectStatus=project.PROJECT_STATUS_ACTIVE
+    ) -> 'Project':
+        req = project.CreateProjectRequest(title=title, access=access, status=status)
+        proj: project.Project = api._proj_stub.Create(req, wait_for_ready=True)
+        return Project(api, proj)
+
+    def delete(self):
+        self.api._proj_stub.Delete(project.DeleteProjectRequest(id=self.id))
+
+    @staticmethod
+    def list_projects(api: 'Api') -> List['Project']:
+        list_req = project.ListProjectRequest(owner_id="")
+        results = []
+        for proj in api._proj_stub.List(list_req, wait_for_ready=True):
+            results.append(Project(api, proj))
+        return results
+
+    def load_settings(self) -> dict:
+        request = generation.Request(
+            engine_id=self.api._asset.engine_id,
+            prompt=[generation.Prompt(
+                artifact=generation.Artifact(
+                    type=generation.ARTIFACT_TEXT,
+                    mime="application/json",
+                    uuid=self.file_id,
+                )
+            )],
+            asset=generation.AssetParameters(
+                action=generation.ASSET_GET, 
+                project_id=self.id,
+                use=generation.ASSET_USE_PROJECT
+            )
+        )
+        for resp in self.api._asset.stub.Generate(request, wait_for_ready=True):
+            for artifact in resp.artifacts:
+                if artifact.type == generation.ARTIFACT_TEXT:
+                    return json.loads(artifact.text)
+        raise Exception(f"Failed to load project file for {self.id}")
+
+    def save_settings(self, data: dict) -> str:
+        contents = json.dumps(data)
+        request = generation.Request(
+            engine_id=self.api._asset.engine_id,
+            prompt=[generation.Prompt(
+                artifact=generation.Artifact(
+                    type=generation.ARTIFACT_TEXT,
+                    text=contents,
+                    mime="application/json",
+                    uuid=self.file_id
+                )
+            )],
+            asset=generation.AssetParameters(
+                action=generation.ASSET_PUT, 
+                project_id=self.id, 
+                use=generation.ASSET_USE_PROJECT
+            )
+        )
+        for resp in self.api._asset.stub.Generate(request, wait_for_ready=True):
+            for artifact in resp.artifacts:
+                if artifact.type == generation.ARTIFACT_TEXT:
+                    self.update(file_id=artifact.uuid, file_uri=artifact.text)
```

Comment:
what's the difference between file_id and file_uri?

Reply:
the file_id is an asset UUID and the file_uri is an https://.... address